sympy/functions/elementary/integers.py

```diff
@@ -76,6 +76,12 @@ def eval(cls, arg):
         elif isinstance(spart, (floor, ceiling)):
             return ipart + spart
         else:
+            numer, denom = spart.as_numer_denom()
+            if denom == 2:
+                if numer.is_even:
+                    return ipart + spart
+                if numer.is_odd:
+                    return ipart + spart + cls._dir/S(2)
             return ipart + cls(spart, evaluate=False)
```

Comment:
The latter is intended. I had expressions like `floor((n + 1)/2)` which will not work with fraction (although `as_numer_denom` was the first I found, so not a deliberate choice).

If I just could figure out how to simplify `2**floor(Nodd/2)` using `_eval_simplify`, I would for sure have put it there... But as far as I can tell, it will select `floor(Nodd/2)` by default, making it a bit useless.

Reply:
Should it not be `floor(Nodd/2) -> (Nodd - 1)/2`?

Reply:
For performance: yes, it is a bit unfortunate. But I restate that since there are quite limited methods to simplify floor and ceiling, I think it is worthwhile if one can get rid of them (although admittedly, it may be a rare case that it covers).

(In my original use case I had a general N, but now I can get two different expressions for odd and even N without explicit  substitutions.)

Reply:
> floor((n + 1)/2)

You could use `fraction(factor_terms((n + 1)/2))`